src/Coordination/KeeperDispatcher.cpp

```diff
@@ -285,6 +296,8 @@ void KeeperDispatcher::initialize(const Poco::Util::AbstractConfiguration & conf
     responses_thread = ThreadFromGlobalPool([this] { responseThread(); });
     snapshot_thread = ThreadFromGlobalPool([this] { snapshotThread(); });
 
+    snapshot_s3.startup();
```

Comment:
> And actually not sure if calling updateConfiguratiton() below should be necessary here.

Yes, it seems it's not necessary but only because we have ConfigReloader, I would leave it as is because ConfigReloader is not part of the KeeperDispatcher.

Reply:
> I guess we should provide initial configuration at startup

What do you mean by the initial configuration?

Reply:
Basically similar to `server`. `config` is provided to its `startup()`.
AFAIU, w/o calling `updateConfiguration()` below, S3 snapshot manager will not create S3 client

Reply:
ah I see, as an argument to startup.
Yes, that makes sense